src/api/DefaultAzureResourceProvider.ts

```diff
@@ -70,3 +79,7 @@ export class DefaultAzureResourceProvider implements AzureResourceProvider {
         };
     }
 }
+
+function dedupeAzureResources(array: GenericResource[]): GenericResource[] {
+    return [...new Map(array.map((item) => [item.id, item])).values()];
```

Comment:
Might make sense to capture the telemetry in here. I know we have to pass more args then to do the calculation, but I think it would be nice to have it all encapsulated in this function especially if we ever end up using it in other places.

Reply:
I think I'll make a function that lists the resources, dedupes them, and also records telemetry.